When a user leaves a group, remove them from all group challenges

colegleason opened the pull request:
This should fix #2195.

However, what should happen when the user is the leader of the challenge?

colegleason committed:
When a user leaves a group, remove them from all group challenges

lefnire commented on public/js/controllers/groupsCtrl.js:
```diff
@@ -238,11 +238,12 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Groups', '$http', 'A
-        Groups.Group.leave({gid: group._id}, undefined, function(){
+     $scope.leave = function(keep) {
+         if (keep == 'cancel') {
+             $scope.selectedChal = undefined;
```
do we want it to be `selectedGroup`?

colegleason replied:
Yup, whoops,

lefnire commented on public/js/controllers/groupsCtrl.js:
```diff
@@ -256,6 +257,23 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Groups', '$http', 'A
+              title: 'Leaving group challenges and...',
+              content: html
+          }).popover('show');
+      }
```
One issue is this will show for user even if they don't belong to any challenges within the group. I anticipate a few bug reports as a result, I'm not quite sure a workaround for it. *starts thinking*

colegleason replied:
Well, we could fetch the data to see if they are in a group's challenge first (I think). That just seemed like a lot of extra computation to me.

lefnire replied:
so there's `user.challenges`, which appears to be being managed on leaving/joining challenges (don't know if we're otherwise using it in any way though). We could do

``` javacript
if (!_.isEmpty(_.intersection(_.pluck(group.challenges,'id'),user.challenges)))
 ...
```

Do you think it's worthwhile? Or do think you think maybe we should just update the popover wording something like "if you haven't subscribed to any challenges, click anything"

colegleason replied:
There isn't a noticeable lag in my new commit. However, group.challenges doesn't seem to be populated, so I fetched the challenges and filtered. Not sure how that might behave with a larger system.

colegleason committed:
Fixed bug where user wasn't really removed. Changed popup depending on if user was in challenges. Fixed selectedChal typo

colegleason commented:
Try this out. I added the fetching of challenges to determine popup content. Also fixed a bug where I wasn't actually removing the challenge from user.challenges :-/

lefnire commented:
Awesome man! I'll take a stab in the morn, cashing in - thanks for working on this!
